pkg/v1/sdk/features/runner/runner_test.go

```diff
@@ -0,0 +1,443 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package runner
+
+import (
+	"context"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	configv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/config/v1alpha1"
+)
+
+func TestGetFeature(t *testing.T) {
+	objs, features, _ := getTestObjects()
+
+	ctx := context.Background()
+	s := scheme.Scheme
+	if err := configv1alpha1.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add config scheme: (%v)", err)
+	}
+	cl := fake.NewClientBuilder().WithRuntimeObjects(objs...).Build()
+	fgr := FeatureGateRunner{featureGateClient: cl}
+
+	getFeatureTestCases := []struct {
+		description string
+		featureName string
+		returnErr   bool
+	}{
+		{
+			description: "should successfully return feature",
+			featureName: "cloud-event-listener",
+			returnErr:   false,
+		},
+		{
+			description: "should return an error when querying for feature that doesn't exist",
+			featureName: "xyz",
+			returnErr:   true,
+		},
+	}
+
+	for _, tc := range getFeatureTestCases {
+		t.Run(tc.description, func(t *testing.T) {
+			feature, err := fgr.getFeature(ctx, tc.featureName)
+			if err != nil {
+				if !tc.returnErr {
+					t.Errorf("error not expected, but got error: %v", err)
+				}
+			} else if tc.returnErr {
+				if err == nil {
+					t.Errorf("error expected, but got nothing")
+				}
+			}
+			if feature != nil && (feature.Name != features[tc.featureName].Name ||
+				feature.Spec.Immutable != features[tc.featureName].Spec.Immutable ||
+				feature.Spec.Discoverable != features[tc.featureName].Spec.Discoverable ||
+				feature.Spec.Activated != features[tc.featureName].Spec.Activated ||
+				feature.Spec.Maturity != features[tc.featureName].Spec.Maturity ||
+				feature.Spec.Description != features[tc.featureName].Spec.Description) {
+				t.Errorf("feature returned is not the correct feature")
```

Comment:
Can this indicate what the correct value should be? Something like a diff of objects. If this test fails, we'd never know what the incorrect value was to fix it.